src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -76,7 +76,7 @@ public static FormulaValue Text_UO(IRContext irContext, UntypedObjectValue[] arg
                 return new StringValue(irContext, str);
             }
 
-            return CommonErrors.RuntimeTypeMismatch(irContext);
+            return GetTypeMismatchError(irContext, "Text", "String", impl);
```

Comment:
These should probably be `BuiltinFunctionsCore.Text.Name` instead of string literals

Reply:
For types, I'd suggest using `DType.<string|number|...>.GetKindString()`